flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/FunctionGenerator.scala

```diff
@@ -784,6 +784,11 @@ class FunctionGenerator private(config: TableConfig) {
     Seq(FLOAT, INTEGER),
     BuiltInMethods.TRUNCATE_FLOAT)
 
+  addSqlFunctionMethod(JSON_EXISTS, Seq(CHAR, CHAR), BuiltInMethods.JSON_EXISTS)
```

Comment:
We don't really care about the specific argument types, only about the number of arguments. However, `FunctionGenerator` uses a map of function + argument types for the lookup, so we need to list the combinations. We discussed extending `FunctionGenerator` to use a map with matchers instead of a fixed set. I'm happy to make that change as a separate commit if needed.

Reply:
Discussed offline that we will not do this for now.